src/spell/MigrationSpell-mainnet.sol

```diff
@@ -0,0 +1,154 @@
+    //     0xd595E1483c507E74E2E6A3dE8e7D08d8f6F74936
+    // ];
+
+    address[] public memberlistMembers = [
+        0xd595E1483c507E74E2E6A3dE8e7D08d8f6F74936,
+        0xeF08Bb6F5F9494faf2316402802e54089E6322eb,
+        0x30d3bbAE8623d0e9C0db5c27B82dCDA39De40997,
+        0xeEDC395aAAb05e5fb6130A8C5AEbAE48E7739B78,
+        0x2923c1B5313F7375fdaeE80b7745106deBC1b53E,
+        0x14FFe68D005e58f08c27dC0c999f75639682276c,
+        0x86552B8d4F4a600D92d516eE8eA8B922EFEcB561
+    ];
+
+    mapping(address => uint64) public validUntil;
+    bool public done;
+    uint256 constant ONE = 10 ** 27;
+    address self;
+
+    constructor() {
+        validUntil[0xd595E1483c507E74E2E6A3dE8e7D08d8f6F74936] = type(uint64).max;
+        validUntil[0xeF08Bb6F5F9494faf2316402802e54089E6322eb] = 2017150288;
+        validUntil[0x30d3bbAE8623d0e9C0db5c27B82dCDA39De40997] = 2017323212;
+        validUntil[0xeEDC395aAAb05e5fb6130A8C5AEbAE48E7739B78] = 3745713365;
+        validUntil[0x2923c1B5313F7375fdaeE80b7745106deBC1b53E] = 2031229099;
+        validUntil[0x14FFe68D005e58f08c27dC0c999f75639682276c] = 2035299660;
+        validUntil[0x86552B8d4F4a600D92d516eE8eA8B922EFEcB561] = 2037188261;
+    }
+
+    function cast() public {
+        require(!done, "spell-already-cast");
+        done = true;
+        execute();
+    }
+
+    function execute() internal {
+        self = address(this);
+        IRoot rootOld = IRoot(address(ROOT_OLD));
+        IRoot rootNew = IRoot(address(ROOT_NEW));
+        IPoolManager poolManager = IPoolManager(address(POOLMANAGER));
+        ITranche trancheTokenOld = ITranche(TRANCHE_TOKEN_OLD);
+        IInvestmentManager investmentManagerOld = IInvestmentManager(address(INVESTMENTMANAGER_OLD));
+        rootOld.relyContract(address(investmentManagerOld), self);
+
+        // deploy new tranche token
+        rootNew.relyContract(address(POOLMANAGER), self);
+        poolManager.addPool(POOL_ID);
+        poolManager.addTranche(POOL_ID, TRANCHE_ID, NAME, SYMBOL, DECIMALS, RESTRICTIONMANAGER);
+        poolManager.addAsset(CURRENCY_ID, CURRENCY);
+        poolManager.allowAsset(POOL_ID, CURRENCY_ID);
+        trancheTokenNew = ITranche(poolManager.deployTranche(POOL_ID, TRANCHE_ID));
+        rootNew.relyContract(address(trancheTokenNew), self);
+
+        // add all old members to new memberlist and claim any tokens
+        uint256 holderBalance;
+        uint256 escrowBalance = trancheTokenOld.balanceOf(ESCROW_OLD);
+        for (uint8 i; i < memberlistMembers.length; i++) {
+            // add member to new memberlist
+            poolManager.updateRestriction(
+                POOL_ID,
+                TRANCHE_ID,
+                abi.encodePacked(
+                    uint8(RestrictionUpdate.UpdateMember),
+                    address(memberlistMembers[i]).toBytes32(),
+                    validUntil[memberlistMembers[i]]
+                )
+            );
+
+            if (escrowBalance > 0) {
```

Comment:
Would be better to. check `maxMint` here and only call `mint()` if `maxMint > 0`